indico/modules/events/registration/fields/simple.py

```diff
@@ -571,30 +572,62 @@ def process_form_data(self, registration, value, old_data=None, billable_items_l
         # always store no-option as empty dict
         if value is None:
             value = {}
+
+        return_value = {}
+
+        if old_data:
+            old_choices_mapping = {x['id']: x for x in old_data.field_data.versioned_data['choices']}
+            new_choices_mapping = {x['id']: x for x in self.form_item.versioned_data['choices']}
+            revoked_selected_choice_ids = []
+            selected_new_option_exists = False
```

Comment:
selected_new_option_exists = value.viewkeys() > old_choices_mapping.viewkeys()

*value's keys are a strict superset of old_choices_mapping's keys* (i.e. there is at least one value that didn't exist in the old choices mapping)